rayon-core/src/thread_pool/mod.rs

```diff
@@ -20,9 +20,8 @@ impl ThreadPool {
     /// Constructs a new thread pool with the given configuration. If
     /// the configuration is not valid, returns a suitable `Err`
     /// result.  See `InitError` for more details.
-    pub fn new(configuration: Configuration) -> Result<ThreadPool, InitError> {
-        try!(configuration.validate());
-        Ok(ThreadPool { registry: Registry::new(configuration) })
+    pub fn new(configuration: Configuration) -> ThreadPool {
+        ThreadPool { registry: Registry::new(configuration) }
```

Comment:
I hesitate to make this infallible.  There's an open question in #205, what to do when thread creation fails.  We could just decide to make that panic "nicely", cleaning up as much as possible and then continuing the panic, so we'd just unwind though here.  But I think it's also reasonable to return a nice error.  Perhaps we should leave this as an opaque `Box<Error>` for future possibilities.